smart-contracts/contracts/cl-vault/src/contract.rs

```diff
@@ -177,13 +190,152 @@ pub fn reply(deps: DepsMut, env: Env, msg: Reply) -> Result<Response, ContractEr
         Replies::WithdrawUser => handle_withdraw_user_reply(deps, msg.result),
         Replies::WithdrawMerge => handle_merge_withdraw_reply(deps, env, msg.result),
         Replies::CreatePositionMerge => handle_merge_create_position_reply(deps, env, msg.result),
+        Replies::WithdrawToMigrate => handle_migration_withdrawal_reply(deps, env, msg.result),
+        Replies::CreateMigratedPosition => {
+            handle_create_migrated_position_reply(deps, env, msg.result)
+        }
         Replies::Unknown => unimplemented!(),
     }
 }
 
 #[cfg_attr(not(feature = "library"), entry_point)]
-pub fn migrate(deps: DepsMut, _env: Env, _msg: MigrateMsg) -> Result<Response, ContractError> {
-    set_contract_version(deps.storage, CONTRACT_NAME, CONTRACT_VERSION)?;
+pub fn migrate(deps: DepsMut, env: Env, msg: MigrateMsg) -> Result<Response, ContractError> {
+    // Step 1: Withdraw current position from the existing pool
+    let position_breakdown = get_position(deps.storage, &deps.querier)?;
+    let position = position_breakdown.position.unwrap();
+
+    let migration_data = MigrationData {
+        new_pool_id: msg.pool_id,
+        lower_tick: msg.lower_tick,
+        upper_tick: msg.upper_tick,
+    };
+
+    MIGRATION_DATA.save(deps.storage, &migration_data)?;
+
+    let withdraw_msg = MsgWithdrawPosition {
+        position_id: position.position_id,
+        sender: env.contract.address.to_string(),
+        liquidity_amount: Decimal256::from_str(position.liquidity.as_str())?
```

Comment:
we need to double check that atomics is what we want here

Reply:
thats what we do in range flow so should be same imo